doc/Caliptra.md

```diff
@@ -1226,20 +1248,199 @@ This section describes Caliptra error reporting and handling.
+*Figure: Caliptra security subsystem*
+![](./images/Subsystem.png)
+
+**Caliptra Subsystem Architectural Flows**
+
+**Subsystem (Pre-FW Load) Boot Flow **
+
+**Note:** Any step done by MCU HW/ROM would have been performed by “SOC Manager” in Caliptra 1p0.
+
+1. SoC (using its HW or MCU ROM) performs pre-steps like bringing up CRO or PLL, MBIST flows on SRAMs, SRAM Init etc.
+2. SOC will assert MCU & Caliptra pwrgood and after 10 cycles MCU
+3. MCU ROM or SOC Manager wrapper will bring up fuse controller and any other SOC specific infrastructure RTL modules (I3C, GPIO programming, Glitch detector programming etc.)
+4. MCU ROM or SOC Manager wrapper will deassert Caliptra reset.
+5. Caliptra HW will read the security centric (secret) fuses.
+6. MCU ROM waits for ready_for_fuses to be asserted.
+7. MCU ROM or SOC Manager will populate the remaining fuses of Caliptra and reads its own fuses (if any). Note that this step is gated behind the completion of security fuse writes to ensure the step has completed.
+8. MCU ROM will write “fuse done” to Caliptra.
+9. Caliptra will go through its boot flow of bringing up uC.
+10. Caliptra ROM starts and executes various KATs flows.
+
+**Subsystem Boot Flow**
+
+**_If (Caliptra-Passive-Mode)_**
+
+1. SOC Manager goes through legacy flows => loads Caliptra FW using legacy flows, Caliptra sets RT ready and SOC  <-> Caliptra boot flow is done.
+   
+**_(Caliptra-Active-Mode)_**
+
+1. Caliptra ROM waits for SOC infrastructure readiness indication. If this indication is set, Caliptra will do the identity derviationderivation flows. If it is not set, then this flow is run when the SOC infrastructure readiness indication is set.
+2. Caliptra ROM will follow the recovery interface protocol to load its FW. Please see the specific section for next level specifics; At a high level, Caliptra ROM sets the device ready in the I3C controller and poll I3C for the payloads.
+3. BMC or a similar platform component will send the image (code or data) through OCP recovery flow protocol.
+   a. Caliptra ROM should implement a recovery capability to allow for BMC to send ‘data’ instead of ‘code’ as a SOC specific configuration OR allow MCU ROM to send some data to be either integrity checked or authenticated. The data flow and code flow over recovery interface is the same from physical interface point of view and follows the recovery spec as implemented in Caliptra subsystem documentation (please see the recovery section).
+   b. This need for data flow (from flash or BMC) is indicated by a SOC configuration bit to Caliptra ROM.
+   c. This ‘data’ flow is possible only before SOC infra readiness is set. This is specifically used for scenarios where PUF or other characterization data is coming off-chip (say a flash or BMC)
```

Comment:
this provides a glitch hook for UDS seed loading, seems dangerous.

Reply:
+1

Reply:
I agree - this is where device specific content/recovery from platform  (eg. PUF characterization data or PQC related fuse content etc.) get nasty - without them we will have chicken and egg problem and all the more reason to figure out a solution without platform dependencies :)

Reply:
I think this needs to be tied to an explicit requirement, because @Bryankel had previously argued against such hooks in the past for this reason.

Reply:
@varuns-nvidia - I think this is what we should do to resolve the spec; we add this as a config, leave a FIXME to analyze security around this before finalizing this.

I am actually 100% aligned that there should be no SOC specific data that is needed for RoT functionality stored on the platform to make it operational (eg. PUF characterization data, PQC signature etc.); All that data is within the SOC "package"

Reply:
Added this - "**FIXME:** Security and operations impact of this step/flow is being analyzed. This capability/flexibility will be updated or removed once that is finalized."